### ISSUE TITLE
Linting single file does not report arity errors for functions from another file called within a `def` block

### ISSUE THREAD

**Issue by MikeJCusack** [clj-kondo/clj-kondo#2641]
When running `clj-kondo --lint src/` it will successfully report the error. When running `clj-kondo --lint src/bar.clj` it will not report the error. This appears to be specific to `def` forms as if you change `baz` to be a `defn` then both commands successfully report the error.

Version: v2025.09.22
Repro:
```clojure
;; foo.clj
(ns foo)

(defn foo [a b]
  (+ a b))

;; bar.clj
(ns bar
  (:require [foo :as foo]))

(def baz (foo/foo 1 2 3)) ;; should report invalid arity
```

**Comment by MikeJCusack** [comment]
Thanks for the quick fix!
